### DISCUSSION TITLE
Lucene search in Headless CMS

### DISCUSSION THREAD

**Discussion by 13Groszy** [OrchardCMS/OrchardCore#14698] [Q&A] [answered]
Hi guys, I believe someone faced this issue before and can help me with it. I've created website using Orchard as data-center with Headless CMS and I retrieve all the pages through GraphQL. All works fine, but I faced a wall with full text search implementation. Is there a chance I can do a full text search there somehow? I've tried to call api/lucene/content but with no results (or I did it incorrectly). Any tips/recommendations will be checked!

Edit: To be precise, I want to create search function to browse all the data from the pages/blogs based on what user type in and return the list of them on the frontend.

Edit2: Enabling Lucene search in the data-center and visiting https://data-center/search?Terms=string always return 'There are no such results.' even tho I've rebuild the lucene indices.

**Comment by jtkech** [comment, marked as answer]
In the settings of a given content type you can define its full text, by code ot through the admin UI.

The full text can include the display text and/or body parts, you can also define a custom full text with liquid syntax so that it can rely on other parts/fields values of a given content item of that type.
